riff-raff/app/notification/ScheduledDeployFailureNotifications.scala

```diff
@@ -0,0 +1,73 @@
+package notification
+
+import java.util.UUID
+import ci.TargetResolver
+import controllers.{Logging, routes}
+import lifecycle.Lifecycle
+import magenta.{DeployParameters, DeployReporter, Fail}
+import scala.concurrent.ExecutionContext
+import com.gu.anghammarad.Anghammarad
+import com.gu.anghammarad.models._
+import conf.Configuration
+import magenta.deployment_type.DeploymentType
+import magenta.input.resolver.Resolver
+import schedule.ScheduledDeployer
+
+class ScheduledDeployFailureNotifications(deploymentTypes: Seq[DeploymentType])(implicit ec: ExecutionContext) extends Lifecycle with Logging {
+
+  private val anghammaradTopicARN = Configuration.scheduledDeployment.anghammaradTopicARN
+  private val snsClient = Configuration.scheduledDeployment.snsClient
+  private val prefix = Configuration.urls.publicPrefix
+
+  def url(uuid: UUID): String = {
+    val path = routes.DeployController.viewUUID(uuid.toString,true)
+    prefix + path.url
+  }
+
+  def failedDeployNotification(uuid: UUID, parameters: DeployParameters) = {
+    val deriveAnghammaradTargets = for {
+      yaml <- TargetResolver.fetchYaml(parameters.build)
+      deployGraph <- Resolver.resolveDeploymentGraph(yaml, deploymentTypes, magenta.input.All).toEither
+    } yield {
+      TargetResolver.extractTargets(deployGraph).toList.flatMap { target =>
+        List(App(target.app), Stack(target.stack))
+      } ++ List(Stage(parameters.stage.name))
+    }
+
+    deriveAnghammaradTargets match {
+      case Right(targets) =>
+        val failureMessage = s"Your scheduled deploy failed. This means that your AMI may become out of date. If no successful deploys occur between now and the next scheduled attempt, it will not be retried."
```

Comment:
Is this the right messaging? Is the only use case for scheduled deploys AMI updates?

Reply:
Good point, I've genericised it (and also added some context about the affected project/build/stage).